streampark-flink/streampark-flink-kubernetes/src/main/scala/org/apache/streampark/flink/kubernetes/KubernetesRetriever.scala

```diff
@@ -47,15 +47,24 @@ object KubernetesRetriever extends Logger {
   // see org.apache.flink.configuration.RestOptions.RETRY_MAX_ATTEMPTS
   val FLINK_REST_RETRY_MAX_ATTEMPTS = 2
 
+  private var kubernetesClient: KubernetesClient = _
+
   /** get new KubernetesClient */
   @throws(classOf[KubernetesClientException])
-  def newK8sClient(): KubernetesClient = {
-    new DefaultKubernetesClient()
+  def getK8sClient(): KubernetesClient = {
```

Comment:
There are better ways to implement the singleton pattern in scala:

```scala 
private lazy val kubernetesClient: KubernetesClient = new DefaultKubernetesClient()
def getK8sClient(): KubernetesClient = kubernetesClient
```

Reply:
Ok, thanks for reviewing the code